tests/ImageProcessor.Web.UnitTests/UrlParserUnitTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="QueryParamParserUnitTests.cs" company="James Jackson-South">
+//   Copyright (c) James Jackson-South.
+//   Licensed under the Apache License, Version 2.0.
+// </copyright>
+// <summary>
+//   The query parameter parser unit tests.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ImageProcessor.Web.UnitTests
+{
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Url decoding tests
+    /// </summary>
+    [TestFixture]
+    public class UrlParserUnitTests
+    {
+        [Test]
```

Comment:
Love that you have separated out the method and made it testable. I'm assuming all the urls mentioned in #506 and #487 also pass?